tools/cli/helpers/doc-parser/runner.php

```diff
@@ -61,20 +61,30 @@ function get_html_from_markdown( $file_path ) {
 		dirname( __DIR__, 4 ) . DIRECTORY_SEPARATOR . $file_path
 	);
 
+	$markdown = mb_convert_encoding( $markdown, 'UTF-8' );
+
 	if ( false === $markdown ) {
 		throw new Exception( 'Could not read Markdown from ' . $file_path );
 	}
 
 	$contents = $parser->defaultTransform( $markdown );
 
 	$document = new DOMDocument();
-	$document->loadHTML( $contents );
+	$document->loadHTML(
+		'<!DOCTYPE html><meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>'
+		. $contents
+	);
 
 	$doc_title = $file_path;
 
 	$headers = $document->getElementsByTagName( 'h1' );
 	if ( count( $headers ) ) {
 		$doc_title = $headers[0]->textContent;
+
+		$parent = $headers[0]->parentNode;
+		if ( null !== $parent ) {
+			$parent->removeChild( $headers[0] );
+		}
```

Comment:
Since we [require PHP 8.2 for this code](https://github.com/Automattic/jetpack/blob/c22675a8ee6c0466302a266a9036dfa2712ff11b/tools/cli/helpers/doc-parser/composer.json#L7), you can just do
```suggestion
		$headers[0]->remove();
```
(that method was added in PHP 8.0)